### ISSUE TITLE
Issues with types and inheritances of those when used inside FormGroup

### ISSUE THREAD

**Issue by acalvomanxa** [angular/angular#55089]
### Which @angular/* package(s) are the source of the bug?

compiler, forms

### Is this a regression?

Yes

### Description

Hi, I'm having trouble getting this to work. What I want to do is have different interfaces to be used in the generic types.

```
interface iZmerControls{
  codipostalMercaderies_input: FormControl<string>,
  katr10_input: FormControl<boolean>,
  nomMercaderies_input: FormControl<string>,
  poblacioMercaderies_input: FormControl<string>,
  poblacio2Mer_input: FormControl<string>,
  carrerMercaderies_input: FormControl<string>,
  telfMercaderies_input: FormControl<string>,
  mail_zmer_input: FormControl<string>,
  mail_c_zmer_input: FormControl<string>,
  phone_zmer_input: FormControl<string>,
  horariMercaderies_input: FormControl<string>,
  latMercaderies_input: FormControl<string>,
  longMercaderies_input: FormControl<string>,
  horariMercaderies_mini_input: FormControl<string>,
  horariMercaderies_mend_input: FormControl<string>,
  horariMercaderies_tini_input: FormControl<string>,
  horariMercaderies_tend_input: FormControl<string>,
  horariMercaderies_dvini_input: FormControl<string>,
  horariMercaderies_dvend_input: FormControl<string>,
  departament_input0: FormControl<string>,
  function_departament_input0: FormControl<string>,
  nomContacte_input0: FormControl<string>,
  telfContacte_input0: FormControl<string>,
  mailContacte_input0: FormControl<string>,
  pais_input: FormControl<string>,
}

interface iZmerControlsFEAG extends iZmerControls  {
  zmer_obrataller_input: FormControl<string>,
  pontgruaMercaderies_input: FormControl<string>,
  trailerMercaderies_input: FormControl<string>,
}

class parent<T extends iZmerControls & {[key in keyof T]: AbstractControl<any>} = iZmerControls>{

  form:FormGroup<T> = new FormGroup({
    codipostalMercaderies_input: new FormControl('',{nonNullable: true}),
    katr10_input: new FormControl(false,{nonNullable: true}),
    nomMercaderies_input: new FormControl('',{nonNullable: true}),
    poblacioMercaderies_input: new FormControl('',{nonNullable: true}),
    poblacio2Mer_input: new FormControl('',{nonNullable: true}),
    carrerMercaderies_input: new FormControl('',{nonNullable: true}),
    telfMercaderies_input: new FormControl('',{nonNullable: true}),
    mail_zmer_input: new FormControl('',{nonNullable: true}),
    mail_c_zmer_input: new FormControl('',{nonNullable: true}),
    phone_zmer_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_input: new FormControl('',{nonNullable: true}),
    latMercaderies_input: new FormControl('',{nonNullable: true}),
    longMercaderies_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_mini_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_mend_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_tini_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_tend_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_dvini_input: new FormControl('',{nonNullable: true}),
    horariMercaderies_dvend_input: new FormControl('',{nonNullable: true}),
    departament_input0: new FormControl('',{nonNullable: true}),
    function_departament_input0: new FormControl('',{nonNullable: true}),
    nomContacte_input0: new FormControl('',{nonNullable: true}),
    telfContacte_input0: new FormControl('',{nonNullable: true}),
    mailContacte_input0: new FormControl('',{nonNullable: true}),
    pais_input: new FormControl('',{nonNullable: true}),
  })
}
```

### Please provide the exception or error you saw

```true
The error states

Type '{ codipostalMercaderies_input: FormControl<string>; katr10_input: FormControl<boolean>; nomMercaderies_input: FormControl<string>; poblacioMercaderies_input: FormControl<...>; ... 20 more ...; pais_input: FormControl<...>; }' is not assignable to type 'ɵIsAny<T, { [key: string]: AbstractControl<any, any>; }, T>';
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17
```

**Comment by JoostK** [comment]
If you change `new FormGroup(...)` to `new FormGroup<T>(...)` you'll get

```
Argument of type '{ codipostalMercaderies_input: FormControl<string>; katr10_input: FormControl<boolean>; nomMercaderies_input: FormControl<string>; poblacioMercaderies_input: FormControl<...>; ... 20 more ...; pais_input: FormControl<...>; }' is not assignable to parameter of type 'T'.
  'T' could be instantiated with an arbitrary type which could be unrelated to '{ codipostalMercaderies_input: FormControl<string>; katr10_input: FormControl<boolean>; nomMercaderies_input: FormControl<string>; poblacioMercaderies_input: FormControl<...>; ... 20 more ...; pais_input: FormControl<...>; }'.
```

which exposes a real problem with this code, unrelated to the typings of `@angular/forms`: `T` may require additional properties that are not present in the `FormGroup`'s controls object.

The error message isn't great here due to `@angular/core`'s complexity to support typed forms in a backwards compatible way, but it's fundamentally an issue with your code.

**Comment by alxhub** [comment]
As @JoostK explains, he type error is a bit hard to parse here, but is correct.

`parent` declares a generic type `T` with the bound of `extends iZmerControls`. This type could be `iZmerControls`, or it could be some type which extends `iZmerControls` and adds extra controls. For example:

```typescript
interface MyControls extends iZmerControls {
  someName: FormControl<number>;
}

new parent<MyControls>();
```

What TypeScript is telling you is that `new FormGroup({...})` in `parent` can't be proven to satisfy `T`, since it might be missing controls which are present in the extended `T`. For example, it would be missing `someName` in the case of `new parent<MyControls>` above.
